kubert/src/lease.rs

```diff
@@ -0,0 +1,469 @@
+//! A distributed cooperative lock implemntation for Kubernetes
+//!
+//! Applications that manage state in Kubernetes--for instance, those that
+//! update resource statuses, may need to coordinate access to that state so
+//! that only one replica is trying to update resources at a time.
+//!
+//! The module manages [`coordv1::Lease`] resources to ensure that only a single
+//! claimant owns the lease.
+
+use k8s_openapi::{api::coordination::v1 as coordv1, apimachinery::pkg::apis::meta::v1 as metav1};
+use std::sync::Arc;
+use tokio::time::Duration;
+
+/// A Kubernetes `Lease`
+#[cfg_attr(docsrs, doc(cfg(feature = "lease")))]
+pub struct Lease {
+    api: Api,
+    name: String,
+    field_manager: String,
+    state: tokio::sync::Mutex<State>,
+}
+
+/// Configuration used when obtaining a lease.
+#[derive(Clone, Debug, PartialEq, Eq)]
+#[cfg_attr(docsrs, doc(cfg(feature = "lease")))]
+pub struct ClaimParams {
```

Comment:
In this case we don't want this, though, since we allow users to construct the struct directly.
```
error[E0639]: cannot create non-exhaustive struct using struct expression
   --> examples/lease.rs:174:26
    |
174 |               let params = kubert::lease::ClaimParams {
    |  __________________________^
175 | |                 lease_duration,
176 | |                 renew_grace_period,
177 | |             };
    | |_____________^
```